test/e2e/operator_test.go

```diff
@@ -228,3 +233,402 @@ func setImage(deployment *appsv1.Deployment, image string) {
+	upstreamSvc := upstreamService(upstreamPodName, upstreamPodNS)
+	if err := cl.Create(context.TODO(), upstreamSvc); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+	}
+	defer func() {
+		if err := cl.Delete(context.TODO(), upstreamSvc); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+		}
+	}()
+	if err := cl.Get(context.TODO(), types.NamespacedName{upstreamSvc.Namespace, upstreamSvc.Name}, upstreamSvc); err != nil {
+		t.Fatalf("failed to get service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+	}
+	upstreamIP := upstreamSvc.Spec.ClusterIP
+
+	// Update cluster DNS forwarding with the upstream resolver's Service IP address.
+	defaultDNS := &operatorv1.DNS{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift-dns",
+			Name:      "default",
+		},
+	}
+	if err := cl.Get(context.TODO(), types.NamespacedName{defaultDNS.Namespace, defaultDNS.Name}, defaultDNS); err != nil {
+		t.Fatalf("failed to get dns %s/%s: %v", defaultDNS.Namespace, defaultDNS.Name, err)
+	}
+	upstream := operatorv1.Server{
+		Name:  "test",
+		Zones: []string{"foo.com"},
+		ForwardPlugin: operatorv1.ForwardPlugin{
+			Upstreams: []string{upstreamIP},
+		},
+	}
+	defaultDNS.Spec.Servers = []operatorv1.Server{upstream}
+	if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+		t.Fatalf("failed to update dns %s/%s: %v", defaultDNS.Namespace, defaultDNS.Name, err)
+	}
+
+	// Verify that the Corefile of DNS DaemonSet pods have been updated.
+	dsNS := "openshift-dns"
+	dsName := "dns-default"
+	dnsDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: dsNS,
+			Name:      dsName,
+		},
+	}
+	if err := cl.Get(context.TODO(), types.NamespacedName{dnsDaemonSet.Namespace, dnsDaemonSet.Name}, dnsDaemonSet); err != nil {
+		t.Fatalf("failed to get daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	selector, err := metav1.LabelSelectorAsSelector(dnsDaemonSet.Spec.Selector)
+	if err != nil {
+		t.Fatalf("daemonset %s/%s has invalid spec.selector: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	defaultDNSPods := &corev1.PodList{}
+	if err := cl.List(context.TODO(), defaultDNSPods, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+		t.Fatalf("failed to list pods for dns daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	catCmd := []string{"cat", "/etc/coredns/Corefile"}
+	dnsContainer := "dns"
+	for _, pod := range defaultDNSPods.Items {
+		if err := lookForStringInPodExec(pod.Namespace, pod.Name, dnsContainer, catCmd, upstreamIP, 2*time.Minute); err != nil {
+			t.Fatalf("failed to lookup Corefile in pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+	}
+
+	// Create the client Pod.
+	testClient := clientPod("test-client", "default", "openshift/origin-cli:latest", []string{"sleep", "3600"})
```

Comment:
Is it possible to discover the client image that's part of the payload under test?

Reply:
Hmm... @ironcladlou is there a resource I can query to get the client image tag?

Reply:
@ironcladlou I'm now discovering the openshift-cli image through the clusteroperator/dns resource [here](https://github.com/openshift/cluster-dns-operator/pull/168/files#diff-382f5df4828722855ac47b478acdf9f4R365-R386).